# Exploring the Indian Startup Ecosystem: A Data Driven Analysis of Funding Trends and Industry Sectors

## Business Understanding
### Business Scenario
Your team is trying to venture into the Indian start-up
ecosystem. As the data expert of the team, you are to
investigate the ecosystem and propose the best course
of action.

*Analyze funding received by start-ups in India from
2018 to 2021.*
- Separate data for each year of funding will is
provided.
- In these datasets, you'll find the start-ups' details,
the funding amounts received, and the investors'
information.

### Business Objective
The aim of this project is to perform analysis on the Indian start-ups ecosystem and advice stakeholders on which venture to invest in to increase the potential of high profit/income.

In [81]:
#import all necessary libraries

# data manipulation
import pandas as pd
import numpy as np

# data visualization libraries
import matplotlib.pyplot as plt
from plotly import express as px
import seaborn as sns

# statistical libraries
from scipy import stats
import statistics as stat

# database manipulation libraries
import pyodbc
from dotenv import dotenv_values

# hide warnings
import warnings
warnings.filterwarnings("ignore")


 

## Setup Database Connection

In [82]:
# load environment variables
environment_variables = dotenv_values(".env")

# load database configurations
database = environment_variables.get("DB_DATABASENAME")
server = environment_variables.get("DB_SERVER")
username = environment_variables.get("DB_USERNAME")
password = environment_variables.get("DB_PASSWORD")

# database connection string
connection_string = f"DRIVER={{SQL Server}};SERVER={server};DATABASE={database};UID={username};PWD={password}"



In [83]:
# create pyodbc connector
connection = pyodbc.connect(connection_string)

In [84]:
# Loading 2021 dataset from MS SQL server
query_2021 = " SELECT * FROM dbo.LP1_startup_funding2021"
df_2021 = pd.read_sql(query_2021,connection)
df_2021.head()

Company_Brand  Founded HeadQuarter          Sector  \
0  Unbox Robotics   2019.0   Bangalore      AI startup   
1          upGrad   2015.0      Mumbai          EdTech   
2     Lead School   2012.0      Mumbai          EdTech   
3         Bizongo   2015.0      Mumbai  B2B E-commerce   
4        FypMoney   2021.0    Gurugram         FinTech   

                                        What_it_does  \
0  Unbox Robotics builds on-demand AI-driven ware...   
1     UpGrad is an online higher education platform.   
2  LEAD School offers technology based school tra...   
3  Bizongo is a business-to-business online marke...   
4  FypMoney is Digital NEO Bank for Teenagers, em...   

                                            Founders  \
0                        Pramod Ghadge, Shahid Memon   
1  Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                         Smita Deorah, Sumeet Mehta   
3            Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                      Kapil Banwari   

                                        Investor        Amount         Stage  
0                    BEENEXT, Entrepreneur First    $1,200,000  Pre-series A  
1       Unilazer Ventures, IIFL Asset Management  $120,000,000          None  
2               GSV Ventures, Westbridge Capital   $30,000,000      Series D  
3                         CDC Group, IDG Capital   $51,000,000      Series C  
4  Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000          Seed

In [85]:
# Load 2020 dataset from MS SQL Server
query_2020 = "SELECT * FROM dbo.LP1_startup_funding2020"
df_2020 = pd.read_sql(query_2020,connection)
df_2020.head()

Company_Brand  Founded HeadQuarter              Sector  \
0    Aqgromalin   2019.0     Chennai            AgriTech   
1      Krayonnz   2019.0   Bangalore              EdTech   
2  PadCare Labs   2018.0        Pune  Hygiene management   
3         NCOME   2020.0   New Delhi              Escrow   
4    Gramophone   2016.0      Indore            AgriTech   

                                        What_it_does  \
0                       Cultivating Ideas for Profit   
1  An academy-guardian-scholar centric ecosystem ...   
2   Converting bio-hazardous waste to harmless waste   
3                       Escrow-as-a-service platform   
4  Gramophone is an AgTech platform enabling acce...   

                                            Founders  \
0                    Prasanna Manogaran, Bharani C L   
1                   Saurabh Dixit, Gurudutt Upadhyay   
2                                    Ajinkya Dhariya   
3                                      Ritesh Tiwari   
4  Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   

                              Investor    Amount     Stage column10  
0                      Angel investors  200000.0      None     None  
1                      GSF Accelerator  100000.0  Pre-seed     None  
2                       Venture Center       NaN  Pre-seed     None  
3  Venture Catalysts, PointOne Capital  400000.0      None     None  
4  Siana Capital Management, Info Edge  340000.0      None     None

In [86]:
# load 2019 dataset
df_2019 = pd.read_csv("data/startup_funding2019.csv")
df_2019.head()

Company/Brand  Founded HeadQuarter           Sector  \
0  Bombay Shaving      NaN         NaN        Ecommerce   
1       Ruangguru   2014.0      Mumbai           Edtech   
2        Eduisfun      NaN      Mumbai           Edtech   
3        HomeLane   2014.0     Chennai  Interior design   
4        Nu Genes   2004.0   Telangana         AgriTech   

                                        What it does  \
0         Provides a range of male grooming products   
1  A learning platform that provides topic-based ...   
2            It aims to make learning fun via games.   
3              Provides interior designing solutions   
4  It is a seed company engaged in production, pr...   

                                Founders  \
0                     Shantanu Deshpande   
1  Adamas Belva Syah Devara, Iman Usman.   
2                          Jatin Solanki   
3           Srikanth Iyer, Rama Harinath   
4                 Narayana Reddy Punyala   

                                            Investor     Amount($)  \
0                               Sixth Sense Ventures    $6,300,000   
1                                   General Atlantic  $150,000,000   
2     Deepak Parekh, Amitabh Bachchan, Piyush Pandey   $28,000,000   
3  Evolvence India Fund (EIF), Pidilite Group, FJ...   $30,000,000   
4           Innovation in Food and Agriculture (IFA)    $6,000,000   

           Stage  
0            NaN  
1       Series C  
2  Fresh funding  
3       Series D  
4            NaN

In [87]:
# load 2018 dataset
df_2018 = pd.read_csv("D:\Programming Stuffs\DAP(Azubi Africa)\Career Accelerator\Sprint1\Indian_Start_Up_Analysis\data\startup_funding2018.csv")
df_2018.head()

Company Name                                           Industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

  Round/Series       Amount                          Location  \
0         Seed       250000       Bangalore, Karnataka, India   
1         Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2     Series A  ₹65,000,000           Gurgaon, Haryana, India   
3        Angel      2000000       Noida, Uttar Pradesh, India   
4         Seed            —  Hyderabad, Andhra Pradesh, India   

                                       About Company  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...  
2          Leading Online Loans Marketplace in India  
3  PayMe India is an innovative FinTech organizat...  
4  Eunimart is a one stop solution for merchants ...

In [88]:
# concatenated all dataset
data = pd.concat([df_2018,df_2019,df_2020,df_2021],ignore_index=False)
data.head() 

Company Name                                           Industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

  Round/Series       Amount                          Location  \
0         Seed       250000       Bangalore, Karnataka, India   
1         Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2     Series A  ₹65,000,000           Gurgaon, Haryana, India   
3        Angel      2000000       Noida, Uttar Pradesh, India   
4         Seed            —  Hyderabad, Andhra Pradesh, India   

                                       About Company Company/Brand  Founded  \
0  TheCollegeFever is a hub for fun, fiesta and f...           NaN      NaN   
1  A startup which aggregates milk from dairy far...           NaN      NaN   
2          Leading Online Loans Marketplace in India           NaN      NaN   
3  PayMe India is an innovative FinTech organizat...           NaN      NaN   
4  Eunimart is a one stop solution for merchants ...           NaN      NaN   

  HeadQuarter Sector What it does Founders Investor Amount($) Stage  \
0         NaN    NaN          NaN      NaN      NaN       NaN   NaN   
1         NaN    NaN          NaN      NaN      NaN       NaN   NaN   
2         NaN    NaN          NaN      NaN      NaN       NaN   NaN   
3         NaN    NaN          NaN      NaN      NaN       NaN   NaN   
4         NaN    NaN          NaN      NaN      NaN       NaN   NaN   

  Company_Brand What_it_does column10  
0           NaN          NaN      NaN  
1           NaN          NaN      NaN  
2           NaN          NaN      NaN  
3           NaN          NaN      NaN  
4           NaN          NaN      NaN

In [89]:
# save dataframe to csv
data.to_csv("Indian_startup_dataset")

In [90]:
# Renaming columns to lowercase with underscores
data = data.rename(columns=lambda x: x.lower().replace(' ', '_'))
data.head()

company_name                                           industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

  round/series       amount                          location  \
0         Seed       250000       Bangalore, Karnataka, India   
1         Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2     Series A  ₹65,000,000           Gurgaon, Haryana, India   
3        Angel      2000000       Noida, Uttar Pradesh, India   
4         Seed            —  Hyderabad, Andhra Pradesh, India   

                                       about_company company/brand  founded  \
0  TheCollegeFever is a hub for fun, fiesta and f...           NaN      NaN   
1  A startup which aggregates milk from dairy far...           NaN      NaN   
2          Leading Online Loans Marketplace in India           NaN      NaN   
3  PayMe India is an innovative FinTech organizat...           NaN      NaN   
4  Eunimart is a one stop solution for merchants ...           NaN      NaN   

  headquarter sector what_it_does founders investor amount($) stage  \
0         NaN    NaN          NaN      NaN      NaN       NaN   NaN   
1         NaN    NaN          NaN      NaN      NaN       NaN   NaN   
2         NaN    NaN          NaN      NaN      NaN       NaN   NaN   
3         NaN    NaN          NaN      NaN      NaN       NaN   NaN   
4         NaN    NaN          NaN      NaN      NaN       NaN   NaN   

  company_brand what_it_does column10  
0           NaN          NaN      NaN  
1           NaN          NaN      NaN  
2           NaN          NaN      NaN  
3           NaN          NaN      NaN  
4           NaN          NaN      NaN

In [91]:
# convert final dataset from csv to dataframe
data.head()

company_name                                           industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

  round/series       amount                          location  \
0         Seed       250000       Bangalore, Karnataka, India   
1         Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2     Series A  ₹65,000,000           Gurgaon, Haryana, India   
3        Angel      2000000       Noida, Uttar Pradesh, India   
4         Seed            —  Hyderabad, Andhra Pradesh, India   

                                       about_company company/brand  founded  \
0  TheCollegeFever is a hub for fun, fiesta and f...           NaN      NaN   
1  A startup which aggregates milk from dairy far...           NaN      NaN   
2          Leading Online Loans Marketplace in India           NaN      NaN   
3  PayMe India is an innovative FinTech organizat...           NaN      NaN   
4  Eunimart is a one stop solution for merchants ...           NaN      NaN   

  headquarter sector what_it_does founders investor amount($) stage  \
0         NaN    NaN          NaN      NaN      NaN       NaN   NaN   
1         NaN    NaN          NaN      NaN      NaN       NaN   NaN   
2         NaN    NaN          NaN      NaN      NaN       NaN   NaN   
3         NaN    NaN          NaN      NaN      NaN       NaN   NaN   
4         NaN    NaN          NaN      NaN      NaN       NaN   NaN   

  company_brand what_it_does column10  
0           NaN          NaN      NaN  
1           NaN          NaN      NaN  
2           NaN          NaN      NaN  
3           NaN          NaN      NaN  
4           NaN          NaN      NaN

In [92]:
# view the last five rows
data.tail()

company_name industry round/series     amount location about_company  \
1204          NaN      NaN          NaN   $3000000      NaN           NaN   
1205          NaN      NaN          NaN  $20000000      NaN           NaN   
1206          NaN      NaN          NaN  $55000000      NaN           NaN   
1207          NaN      NaN          NaN  $26000000      NaN           NaN   
1208          NaN      NaN          NaN   $8000000      NaN           NaN   

     company/brand  founded headquarter                 sector what_it_does  \
1204           NaN   2019.0    Gurugram  Staffing & Recruiting          NaN   
1205           NaN   2015.0   New Delhi       Food & Beverages          NaN   
1206           NaN   2019.0   Bangalore     Financial Services          NaN   
1207           NaN   2015.0    Gurugram                 EdTech          NaN   
1208           NaN   2019.0   Bangalore     Financial Services          NaN   

                           founders                          investor  \
1204    Chirag Mittal, Anirudh Syal                   Endiya Partners   
1205                     Bala Sarda                          IIFL AMC   
1206     Arnav Kumar, Vaibhav Singh                      Owl Ventures   
1207                   Ruchir Arora  Winter Capital, ETS, Man Capital   
1208  Vishal Chopra, Himanshu Gupta    3one4 Capital, Kalaari Capital   

     amount($)         stage company_brand  \
1204       NaN  Pre-series A      Gigforce   
1205       NaN      Series D        Vahdam   
1206       NaN      Series C  Leap Finance   
1207       NaN      Series B  CollegeDekho   
1208       NaN      Series A        WeRize   

                                           what_it_does column10  
1204                  A gig/on-demand staffing company.      NaN  
1205  VAHDAM is among the world’s first vertically i...      NaN  
1206  International education loans for high potenti...      NaN  
1207  Collegedekho.com is Student’s Partner, Friend ...      NaN  
1208  India’s first socially distributed full stack ...      NaN

In [93]:
# check the shape of data
data.shape

(2879, 18)

In [94]:
# check for columns in the data
data.columns

Index(['company_name', 'industry', 'round/series', 'amount', 'location',
       'about_company', 'company/brand', 'founded', 'headquarter', 'sector',
       'what_it_does', 'founders', 'investor', 'amount($)', 'stage',
       'company_brand', 'what_it_does', 'column10'],
      dtype='object')

## Hypothesis Testing
*Hypothesis* - The amount of funds a company receive depends on the sector a company finds itself
- Null Hypothesis(H_o) - The funds a company receive does not depend on the sector of investment
- Alternate Hypothesis(H_a) - The funds a company receive depends on the sector of investment

### Business Questions
- Which particular sector received the most funding over the time frame?
- The distribution of start ups in stages and the amount allocated each?
- In which 3 locations have start ups had the most funding?
- Which year had the most investors?
- Top 3 investor considerations in investing in start ups
- What was the impact of COVID-19 pandemic on startup funding in 2020?
- How is funding related to Metropolitan cities
- What is the distribution of companies based on the Round/Series 

## Data Understanding
Data Understanding it drives the focus to identify, collect, and analyse
the data sets that can help you accomplish the project goals. This
phase also has four tasks:
### Tasks To Be Performed
1. Collect initial data: Acquire the necessary data and (if
necessary) load it into your analysis tool.
2. Describe data: Examine the data and document its surface
properties like data format, number of records, or field
identities.
3. Explore data: Dig deeper into the data. Query it, visualize it,
and identify relationships among the data.
4. Verify data quality: How clean/dirty is the data? Document
any quality issues.

## Data Preprocessing
Data Preprocessing is done to check the quality of the data.The following methods can be used to check the quality of data
- Data types
- Missing data
- Categorical values
- Continuous variable values
- Duplicated records
- Custom rule based checks

*Note*: After checking for duplicated records, we have to go through the columns and perform rule based checks where appropriate.



# Data Exploration and Cleaning on the 2018 Dataset

In [95]:
df_2018.head()

Company Name                                           Industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

  Round/Series       Amount                          Location  \
0         Seed       250000       Bangalore, Karnataka, India   
1         Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2     Series A  ₹65,000,000           Gurgaon, Haryana, India   
3        Angel      2000000       Noida, Uttar Pradesh, India   
4         Seed            —  Hyderabad, Andhra Pradesh, India   

                                       About Company  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...  
2          Leading Online Loans Marketplace in India  
3  PayMe India is an innovative FinTech organizat...  
4  Eunimart is a one stop solution for merchants ...

In [96]:
# rename column labels
df_2018 = df_2018.rename(columns=lambda x: x.lower().replace(" ","_"))
df_2018.head()

company_name                                           industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

  round/series       amount                          location  \
0         Seed       250000       Bangalore, Karnataka, India   
1         Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2     Series A  ₹65,000,000           Gurgaon, Haryana, India   
3        Angel      2000000       Noida, Uttar Pradesh, India   
4         Seed            —  Hyderabad, Andhra Pradesh, India   

                                       about_company  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...  
2          Leading Online Loans Marketplace in India  
3  PayMe India is an innovative FinTech organizat...  
4  Eunimart is a one stop solution for merchants ...

In [97]:
# rename round/stage column to stage
df_2018 =df_2018.rename(columns={"round/series":"stage"}) 

In [98]:
# check for the columns in the dataframe
df_2018.columns

Index(['company_name', 'industry', 'stage', 'amount', 'location',
       'about_company'],
      dtype='object')

In [99]:
# check the shape of the 2018 dataset
df_2018.shape

(526, 6)

In [100]:
# perform descriptive statistics on the dataset
df_2018.describe(include="all").T

count unique                                                top  \
company_name    526    525                                    TheCollegeFever   
industry        526    405                                                  —   
stage           526     21                                               Seed   
amount          526    198                                                  —   
location        526     50                        Bangalore, Karnataka, India   
about_company   526    524  TheCollegeFever is a hub for fun, fiesta and f...   

              freq  
company_name     2  
industry        30  
stage          280  
amount         148  
location       102  
about_company    2

In [101]:
# check the info about the dataset
df_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   company_name   526 non-null    object
 1   industry       526 non-null    object
 2   stage          526 non-null    object
 3   amount         526 non-null    object
 4   location       526 non-null    object
 5   about_company  526 non-null    object
dtypes: object(6)
memory usage: 24.8+ KB


## Observations on the 2018 dataset
- The dataset consists of 526 rows and 6 columns
- All the data types of the columns are of the object type
- The amount column is supposed to be in float

### Course of Action
- Convert the amount column data type to float
- Drop duplicated rows

In [102]:
# check for null values
df_2018.isna().sum()

company_name     0
industry         0
stage            0
amount           0
location         0
about_company    0
dtype: int64

In [103]:
# check for duplicated values
df_2018.duplicated().sum()

1

In [104]:
# drop the duolicate value
df_2018.drop_duplicates(inplace=True)


In [105]:
# verify if duplicate still exists
df_2018.duplicated().sum() 

0

In [106]:
# create function to clean the amount column
exchange_rate = 1/0.0146
    
def clean_amount(df, col_name):
    cleaned_data = df[col_name].copy()
    if df[col_name].dtype == 'object':
        # Remove non-numeric characters except '.' and ','
        cleaned_data = cleaned_data.str.replace("[^\d.,]", "", regex=True)
        
        # Replace empty strings with NaN
        cleaned_data = cleaned_data.replace('', float('nan'), regex=True)
        
        # Remove commas from numbers
        cleaned_data = cleaned_data.str.replace(",", "", regex=True)
        
        if cleaned_data.str.startswith("$").any():
            # If the column starts with '$', clean it as dollars
            cleaned_data = cleaned_data.str.replace("[$]", "", regex=True).astype(float)
        elif cleaned_data.str.startswith("₹").any():
            # If the column starts with '₹', clean it as rupees
            cleaned_data = cleaned_data.str.replace("[₹]", "", regex=True).astype(float) * exchange_rate
        else:
            # Convert to float if there is no currency symbol preceding the amount
            cleaned_data = cleaned_data.astype(float)
    return cleaned_data



In [107]:
# apply the clean_data function to the 2018 dataframe
df_2018["amount"] = clean_amount(df_2018, "amount")

In [108]:
# check info about the dataset after cleaning
df_2018.info()

<class 'pandas.core.frame.DataFrame'>
Index: 525 entries, 0 to 525
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   company_name   525 non-null    object 
 1   industry       525 non-null    object 
 2   stage          525 non-null    object 
 3   amount         377 non-null    float64
 4   location       525 non-null    object 
 5   about_company  525 non-null    object 
dtypes: float64(1), object(5)
memory usage: 28.7+ KB


In [109]:
# check for the null values present after cleaning the amount column
df_2018.isna().sum()

company_name       0
industry           0
stage              0
amount           148
location           0
about_company      0
dtype: int64

In [110]:
average_amount_by_stage = df_2018.groupby("stage")["amount"].mean()

In [117]:
# Function to fill NaN values in the amount column with the average based on stage of company
def impute_amount_column(df,filter_name,fill_value):
    unique_values = df[filter_name].unique()
    for val, avg_amount in zip(unique_values,df.groupby(filter_name)[fill_value].transform('mean')):
        df.loc[df[filter_name] == val, fill_value] = df.loc[df[filter_name] == val, fill_value].fillna(avg_amount)
    return df
     


In [118]:
df_2018 = impute_amount_column(df_2018,"stage","amount")

In [119]:
df_2018

company_name                                           industry  \
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

        stage        amount                          location  \
0        Seed  2.500000e+05       Bangalore, Karnataka, India   
1        Seed  4.000000e+07        Mumbai, Maharashtra, India   
2    Series A  6.500000e+07           Gurgaon, Haryana, India   
3       Angel  2.000000e+06       Noida, Uttar Pradesh, India   
4        Seed  3.328377e+07  Hyderabad, Andhra Pradesh, India   
..        ...           ...                               ...   
521  Series C  2.250000e+08       Bangalore, Karnataka, India   
522  Series A  3.328377e+07           Haryana, Haryana, India   
523      Seed  7.500000e+03        Mumbai, Maharashtra, India   
524      Seed  3.500000e+07        Mumbai, Maharashtra, India   
525  Series C  3.500000e+07        Chennai, Tamil Nadu, India   

                                         about_company  
0    TheCollegeFever is a hub for fun, fiesta and f...  
1    A startup which aggregates milk from dairy far...  
2            Leading Online Loans Marketplace in India  
3    PayMe India is an innovative FinTech organizat...  
4    Eunimart is a one stop solution for merchants ...  
..                                                 ...  
521  Udaan is a B2B trade platform, designed specif...  
522            HappyEasyGo is an online travel domain.  
523  Mombay is a unique opportunity for housewives ...  
524  Droni Tech manufacture UAVs and develop softwa...  
525       Welcome to India's most convenient pharmacy!  

[525 rows x 6 columns]

In [116]:
# reload the 2018 dataframe to check where the null values are
df_2018.info()

<class 'pandas.core.frame.DataFrame'>
Index: 525 entries, 0 to 525
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   company_name   525 non-null    object 
 1   industry       525 non-null    object 
 2   stage          525 non-null    object 
 3   amount         525 non-null    float64
 4   location       525 non-null    object 
 5   about_company  525 non-null    object 
dtypes: float64(1), object(5)
memory usage: 28.7+ KB


In [33]:
# apply cleaned_data function on dataframe
df_2018 = clean_data(df_2018,"amount")

AttributeError: 'str' object has no attribute 'str'

In [60]:
# check for the currency with the dollar symbol
starts_with_dollar = df_2018["amount"].str.startswith("$")
dollar_data = df_2018[starts_with_dollar]

# count the currencies with dollar infront
len(dollar_data)
    

59

In [61]:
# check for the currency with the Rupee symbol
starts_with_rupee = df_2018["amount"].str.startswith("₹")
rupee_data = df_2018[starts_with_rupee]

# count the currencies with dollar infront
len(rupee_data)

144

In [65]:
# strip off currency symbols from the dataset
rupee_data["amount"] = rupee_data["amount"].str.replace("₹","")

# remove all commas from the column 
rupee_data["amount"] = rupee_data["amount"].str.replace(",","")

# convert the amount column to float
rupee_data["amount"] = rupee_data["amount"].astype(float)
rupee_data.head()

company_name                                          industry  \
1   Happy Cow Dairy                              Agriculture, Farming   
2        MyLoanCare  Credit, Financial Services, Lending, Marketplace   
6         Tripshelf                    Internet, Leisure, Marketplace   
7      Hyperdata.IO                                   Market Research   
15          Pitstop       Automotive, Search Engine, Service Industry   

   round/series       amount                          location  \
1          Seed   40000000.0        Mumbai, Maharashtra, India   
2      Series A   65000000.0           Gurgaon, Haryana, India   
6          Seed   16000000.0             Kalkaji, Delhi, India   
7         Angel   50000000.0  Hyderabad, Andhra Pradesh, India   
15         Seed  100000000.0       Bengaluru, Karnataka, India   

                                        about_company  
1   A startup which aggregates milk from dairy far...  
2           Leading Online Loans Marketplace in India  
6   Tripshelf is an online market place for holida...  
7   Hyperdata combines advanced machine learning w...  
15  Pitstop offers general repair and maintenance ...

In [67]:
# convert Rupees to Dollars
exchange_rate = 0.0146 #0.0146USD = 1 Rupee
rupee_data["amount"] = rupee_data["amount"] * (1/exchange_rate)
rupee_data.head()


company_name                                          industry  \
1   Happy Cow Dairy                              Agriculture, Farming   
2        MyLoanCare  Credit, Financial Services, Lending, Marketplace   
6         Tripshelf                    Internet, Leisure, Marketplace   
7      Hyperdata.IO                                   Market Research   
15          Pitstop       Automotive, Search Engine, Service Industry   

   round/series        amount                          location  \
1          Seed  2.739726e+09        Mumbai, Maharashtra, India   
2      Series A  4.452055e+09           Gurgaon, Haryana, India   
6          Seed  1.095890e+09             Kalkaji, Delhi, India   
7         Angel  3.424658e+09  Hyderabad, Andhra Pradesh, India   
15         Seed  6.849315e+09       Bengaluru, Karnataka, India   

                                        about_company  
1   A startup which aggregates milk from dairy far...  
2           Leading Online Loans Marketplace in India  
6   Tripshelf is an online market place for holida...  
7   Hyperdata combines advanced machine learning w...  
15  Pitstop offers general repair and maintenance ...

In [70]:
# combine the dollar and the rupee dataset 
df_2018 = pd.concat([dollar_data,rupee_data],ignore_index=False)
df_2018.head()

company_name  \
86                                WHR   
90                           SBI Life   
93   NoPaperForms Solutions Pvt. Ltd.   
95                         AuthMetrik   
101                            Swiggy   

                                              industry    round/series  \
86                 Health Care, Information Technology            Seed   
90                                           Insurance  Private Equity   
93       EdTech, Education, Information Services, SaaS        Series B   
95   B2B, Biometrics, Cyber Security, Fraud Detecti...           Grant   
101           Food Delivery, Food Processing, Internet        Series H   

           amount                     location  \
86   1.431450e+05     Pune, Maharashtra, India   
90   7.420000e+08   Mumbai, Maharashtra, India   
93   3.980000e+06      New Delhi, Delhi, India   
95   1.000000e+04      Gurgaon, Haryana, India   
101  1.000000e+09  Bangalore, Karnataka, India   

                                         about_company  
86   WHR is to make affordable healthcare a reality...  
90   SBI Life is one of the life insurance company ...  
93   NoPaperForms is a marketing automation, lead n...  
95   SaaS, B2B, Security, Stop account sharing, Fra...  
101  Swiggy is a food ordering and delivery company...

In [72]:
# check the information on the dataset
df_2018.shape

(203, 6)

In [ ]:
# Clean the amount column inside the rupee data
# strip off currency symbols from the dataset
dollar_data["amount"] = dollar_data["amount"].str.replace("$","")

# remove all commas from the column 
dollar_data["amount"] = dollar_data["amount"].str.replace(",","")

# convert the amount column to float
dollar_data["amount"] = dollar_data["amount"].astype(float)
dollar_data.head()

From the analysis of the currencies in the amoount column, we have 144 amount in Rupees and 59 data with the dollar sign 

In [30]:
# convert the data type of amount to float
df_2018["amount"] = df_2018["amount"].astype(float) 

ValueError: could not convert string to float: '₹40,000,000'

In [11]:
# perform descriptive statistics on data
data.describe(include="all").T

count unique  \
company_name      526    525   
industry          526    405   
round/series      526     21   
amount           2533    754   
location          526     50   
about_company     526    524   
company/brand      89     87   
founded        2110.0    NaN   
headquarter      2239    123   
sector           2335    502   
what_it_does       89     88   
founders         2334   1980   
investor         2253   1777   
amount($)          89     50   
stage            1415     62   
company_brand    2264   1745   
what_it_does     2264   2112   
column10            2      2   

                                                             top freq  \
company_name                                     TheCollegeFever    2   
industry                                                       —   30   
round/series                                                Seed  280   
amount                                                         —  148   
location                             Bangalore, Karnataka, India  102   
about_company  TheCollegeFever is a hub for fun, fiesta and f...    2   
company/brand                                           Kratikal    2   
founded                                                      NaN  NaN   
headquarter                                            Bangalore  764   
sector                                                   FinTech  173   
what_it_does                                    Online meat shop    2   
founders                                         Byju Raveendran    7   
investor                               Inflection Point Ventures   36   
amount($)                                            Undisclosed   12   
stage                                                       Seed  326   
company_brand                                           BharatPe   10   
what_it_does   BYJU'S is an educational technology company th...    5   
column10                                                Pre-Seed    1   

                      mean       std     min     25%     50%     75%     max  
company_name           NaN       NaN     NaN     NaN     NaN     NaN     NaN  
industry               NaN       NaN     NaN     NaN     NaN     NaN     NaN  
round/series           NaN       NaN     NaN     NaN     NaN     NaN     NaN  
amount                 NaN       NaN     NaN     NaN     NaN     NaN     NaN  
location               NaN       NaN     NaN     NaN     NaN     NaN     NaN  
about_company          NaN       NaN     NaN     NaN     NaN     NaN     NaN  
company/brand          NaN       NaN     NaN     NaN     NaN     NaN     NaN  
founded        2016.079621  4.368006  1963.0  2015.0  2017.0  2019.0  2021.0  
headquarter            NaN       NaN     NaN     NaN     NaN     NaN     NaN  
sector                 NaN       NaN     NaN     NaN     NaN     NaN     NaN  
what_it_does           NaN       NaN     NaN     NaN     NaN     NaN     NaN  
founders               NaN       NaN     NaN     NaN     NaN     NaN     NaN  
investor               NaN       NaN     NaN     NaN     NaN     NaN     NaN  
amount($)              NaN       NaN     NaN     NaN     NaN     NaN     NaN  
stage                  NaN       NaN     NaN     NaN     NaN     NaN     NaN  
company_brand          NaN       NaN     NaN     NaN     NaN     NaN     NaN  
what_it_does           NaN       NaN     NaN     NaN     NaN     NaN     NaN  
column10               NaN       NaN     NaN     NaN     NaN     NaN     NaN

In [12]:
# check the information about the dataset
data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2879 entries, 0 to 1208
Data columns (total 18 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   company_name   526 non-null    object 
 1   industry       526 non-null    object 
 2   round/series   526 non-null    object 
 3   amount         2533 non-null   object 
 4   location       526 non-null    object 
 5   about_company  526 non-null    object 
 6   company/brand  89 non-null     object 
 7   founded        2110 non-null   float64
 8   headquarter    2239 non-null   object 
 9   sector         2335 non-null   object 
 10  what_it_does   89 non-null     object 
 11  founders       2334 non-null   object 
 12  investor       2253 non-null   object 
 13  amount($)      89 non-null     object 
 14  stage          1415 non-null   object 
 15  company_brand  2264 non-null   object 
 16  what_it_does   2264 non-null   object 
 17  column10       2 non-null      object 
dtypes: float64(1)


### Observations
- The dataset consists of 18 columns in all with some duplicated columns
- It is assumed that the duplicated columns are; what_it_does, company_brand and company/brand
- The datatype of 17 of the columns are objects except the founded column which is a float 
- All the columns in the dataset consists of null values
- Column 10 has no real signinficant data and will be dropped in data cleaning
- Some rows have wrong information pertaining to the columns and will have to be discussed further
- Nulls in the dataset will also have to be discussed and dealt with


### Asumptions
- All amounts will be converted to USD in the Data Cleaning and EDA
 - 2018 amount column will assume the currency of USD
- Rename of df_2018 columns based on similarities in wording and comparison with other years


